src/lib/core/components/visualizations/implementations/MosaicVisualization.tsx

```diff
@@ -240,15 +248,22 @@ function MosaicViz(props: Props) {
         vizConfig.xAxisVariable,
         vizConfig.yAxisVariable
       );
-      const response = isTwoByTwo
-        ? dataClient.getTwoByTwo(
-            computation.type,
-            params as MosaicRequestParams
-          )
-        : dataClient.getMosaic(computation.type, params as MosaicRequestParams);
-      return isTwoByTwo
-        ? twoByTwoResponseToData(await response)
-        : mosaicResponseToData(await response);
+
+      if (isTwoByTwo) {
+        const response = dataClient.getTwoByTwo(
+          computation.type,
+          params as MosaicRequestParams
+        );
+
+        return twoByTwoResponseToData(await response);
+      } else {
+        const response = dataClient.getMosaic(
+          computation.type,
+          params as MosaicRequestParams
```

Comment:
I don't think this is necessary
```suggestion
          params
```

Reply:
Resolved